web/src/engine/websites/Delitoon.ts

```diff
@@ -0,0 +1,174 @@
+import { FetchJSON } from '../platform/FetchProvider';
+import { Exception } from '../Error';
+import { WebsiteResourceKey as R } from '../../i18n/ILocale';
+
+export type APIResult<T> = {
+    result: string,
+    data: T
+}
+
+export type APIManga = {
+    id: number,
+    alias: string,
+    title: string
+    episodes: APIChapter[]
+}
+
+type APIChapter = {
+    id: string,
+    alias: string,
+    title: string,
+    subTitle: string
+}
+
+export type APIPages = {
+    images: {
+        imagePath: string
+    }[]
+}
+
+//Result from /auth/session : to get token
+type APIUser = {
+    user?: APIToken
+}
+
+//Result from /auth/refresh : to refresh token
+type APIRefreshToken = {
+    result: APIToken
+}
+
+//Auth token entity
+type APIToken = {
+    accessToken: {
+        token: string,
+        expiredAt: number
+    },
+    refreshToken: {
+        token: string
+    }
+}
+
+@Common.ImageAjax()
+export default class extends DecoratableMangaScraper {
+    protected BalconyID: string = undefined;
+    protected Timezone: string = 'Europe/Paris';
+    protected Platform: string = 'WEB';
+    protected Token: APIToken = undefined;
+
+    public constructor(id = 'delitoon', label = 'Delitoon', url = 'https://www.delitoon.com', balconyID = 'DELITOON_COM', tags = [Tags.Media.Manhwa, Tags.Language.French, Tags.Source.Official]) {
+        super(id, label, url, ...tags);
+        this.BalconyID = balconyID;
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return new RegExp(`^${this.URI.origin}/detail/[^/]+$`).test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const mangaid = new URL(url).href.match(/\/detail\/([^/]+)/)[1];
+        const req = new URL(`/api/balcony-api-v2/contents/${mangaid}`, this.URI);
+        req.searchParams.set('isNotLoginAdult', 'true');
+        const { data } = await FetchJSON<APIResult<APIManga>>(this.CreateRequest(req));
+        return new Manga(this, provider, mangaid, data.title.trim());
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const url = new URL('/api/balcony-api-v2/contents/search', this.URI);
+        url.search = new URLSearchParams({
+            searchText: '',
+            isCheckDevice: 'true',
+            isIncludeAdult: 'true',
+            contentsThumbnailType: 'MAIN'
+        }).toString();
+        const { data } = await FetchJSON<APIResult<APIManga[]>>(this.CreateRequest(url));
+        return data.map(element => new Manga(this, provider, element.alias, element.title.trim()));
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        await this.GetToken();
+        const url = new URL(`/api/balcony-api-v2/contents/${manga.Identifier}`, this.URI);
+        url.searchParams.set('isNotLoginAdult', 'true');
```

Comment:
Does this exclude all adult content?
How is adult content accessible.

Reply:
No. From what i see, its a variable to get adult content without being logged.